src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -251,13 +251,29 @@ bool CompositionalCouplingScheme::isActionRequired(
   return isRequired;
 }
 
+bool CompositionalCouplingScheme::isActionFulfilled(
+    const std::string &actionName) const
+{
+  PRECICE_TRACE(actionName);
+  bool isFulfilled = false;
+  for (const Scheme &scheme : _couplingSchemes) {
+    if (not scheme.onHold) {
+      isFulfilled |= scheme.scheme->isActionRequired(actionName);
+    }
+  }
+  PRECICE_DEBUG("return {}", isFulfilled);
+  return isFulfilled;
```

Comment:
This looks like there is a "not" missing.
Or did you want to call `isActionFulfilled`?